lib/Segments/DynamicSegments/Filters/WooCommerceCountry.php

```diff
@@ -14,33 +16,70 @@ class WooCommerceCountry implements Filter {
   /** @var EntityManager */
   private $entityManager;
 
+  /** @var DBCollationChecker */
+  private $collationChecker;
+
   public function __construct(
-    EntityManager $entityManager
+    EntityManager $entityManager,
+    DBCollationChecker $collationChecker
   ) {
     $this->entityManager = $entityManager;
+    $this->collationChecker = $collationChecker;
   }
 
   public function apply(QueryBuilder $queryBuilder, DynamicSegmentFilterEntity $filter): QueryBuilder {
     global $wpdb;
     $filterData = $filter->getFilterData();
-    $countryCode = (string)$filterData->getParam('country_code');
-    $countryFilterParam = 'countryCode' . $filter->getId() ?? Security::generateRandomString();
+    $countryCode = $filterData->getParam('country_code');
+    if (!is_array($countryCode)) {
+      $countryCode = [(string)$countryCode];
+    }
+    $operator = $filterData->getParam('operator');
+    if (!$operator) {
+      $operator = DynamicSegmentFilterData::OPERATOR_ANY;
+    }
+
+    $countryFilterParam = ((string)$filter->getId()) . Security::generateRandomString();
+    $condition = $this->createCondition($countryCode, $operator, $countryFilterParam);
     $subscribersTable = $this->entityManager->getClassMetadata(SubscriberEntity::class)->getTableName();
-    return $queryBuilder->innerJoin(
+    $collation = $this->collationChecker->getCollateIfNeeded(
       $subscribersTable,
-      $wpdb->postmeta,
-      'postmeta',
-      "postmeta.meta_key = '_customer_user' AND $subscribersTable.wp_user_id=postmeta.meta_value"
-    )->innerJoin(
-      'postmeta',
-      $wpdb->posts,
-      'posts',
-      "postmeta.post_id = posts.id AND posts.post_status NOT IN ('wc-cancelled', 'wc-failed')"
+      'email',
+      $wpdb->prefix . 'wc_customer_lookup',
+      'email'
+    );
+    $qb = $queryBuilder->innerJoin(
+      $subscribersTable,
+      $wpdb->prefix . 'wc_customer_lookup',
+      'customer',
+      "$subscribersTable.email = customer.email $collation"
     )->innerJoin(
-      'postmeta',
-      $wpdb->postmeta,
-      'postmetaCountry',
-      "postmeta.post_id = postmetaCountry.post_id AND postmetaCountry.meta_key = '_billing_country' AND postmetaCountry.meta_value = :$countryFilterParam"
-    )->setParameter($countryFilterParam, $countryCode);
+      'customer',
+      $wpdb->prefix . 'wc_order_stats',
```

Comment:
A note on using these lookup tables: on my local WooCommerce installation, which is quite old, some of these tables, e.g. `wc_order_stats`, were empty despite me having orders and customers. So a segment was giving me `0` subscribers no matter the settings. I had to submit new orders to see some data put into this table. Then the segment began to find subscribers, but only new ones. Can something akin to this happen with customer installations or is my case just weird?

Reply:
Looks like your tables are out of sync with postmeta.
If you run `wp wc tool run regenerate_product_lookup_tables --user=1` do you get the data populated?

> Can something akin to this happen with customer installations or is my case just weird?

Hopefully, this is not a common scenario. We have started using the tables here to solve performance problems: https://github.com/mailpoet/mailpoet/pull/3862/files

Reply:
> If you run wp wc tool run regenerate_product_lookup_tables --user=1 do you get the data populated?

The command run successfully, but the tables weren't populated with data. 

> Hopefully, this is not a common scenario.

Okay, let's assume this is a problem only with my WC installation, and not a widespread thing. And it is probably related to WooCommerce, not MailPoet.